src/System.Windows.Extensions/tests/System/Drawing/Printing/MarginsConverterTests.cs

```diff
@@ -0,0 +1,172 @@
+
+            IDictionary values = new Dictionary<string, string>();
+            values.Add("Left", "1");
+            values.Add("Right", "2");
+            values.Add("Top", "3");
+            Assert.Throws<ArgumentException>(() => mc.CreateInstance(context, values));
+            values.Add("Bottom", "4");
+            if (PlatformDetection.IsFullFramework)
+            {
+                Assert.Throws<ArgumentException>(() => mc.CreateInstance(context, values));
+                return;
+            }
+            object result = mc.CreateInstance(context, values);
+            Assert.NotNull(result);
+
+            Assert.IsType<Margins>(result);
+            Margins margins = result as Margins;
+            Assert.Equal(1, margins.Left);
+            Assert.Equal(2, margins.Right);
+            Assert.Equal(3, margins.Top);
+            Assert.Equal(4, margins.Bottom);
+        }
+
+        [Fact]
+        public void GetCreateInstanceSupported()
+        {
+            MarginsConverter mc = new MarginsConverter();
+            Assert.True(mc.GetCreateInstanceSupported(null));
+            Assert.True(mc.GetCreateInstanceSupported(new MyTypeDescriptorContext()));
+        }
+
+        [Fact]
+        public void ConvertFrom()
+        {
+            MarginsConverter mc = new MarginsConverter();
+            CultureInfo culture = CultureInfo.InvariantCulture;
+
+            // try once with then once without context
+            for (var context = new MyTypeDescriptorContext(); context != null; context = null)
+            {
+                object result;
+                Assert.Equal(',', culture.TextInfo.ListSeparator[0]);
+                if (PlatformDetection.IsFullFramework)
+                {
+                    Assert.Throws<Exception>(() => mc.ConvertFrom(context, culture, "1;2;3;4"));
+                }
+                result = mc.ConvertFrom(context, culture, "1,2,3,4");
+                Assert.IsType<Margins>(result);
+                Margins margins = result as Margins;
+                Assert.Equal(1, margins.Left);
+                Assert.Equal(2, margins.Right);
+                Assert.Equal(3, margins.Top);
+                Assert.Equal(4, margins.Bottom);
+            }
+        }
+
+        [Fact]
+        public void ConvertFrom_Throws()
+        {
+
+            MarginsConverter mc = new MarginsConverter();
+            CultureInfo culture = CultureInfo.InvariantCulture;
+
+            // try once with then once without context
+            for (var context = new MyTypeDescriptorContext(); context != null; context = null)
+            {
+                Assert.Throws<NotSupportedException>(() => mc.ConvertFrom(context, null, null));
+                Assert.Throws<NotSupportedException>(() => mc.ConvertFrom(context, culture, null));
+                Assert.Throws<NotSupportedException>(() => mc.ConvertFrom(context, culture, Guid.NewGuid()));
+
+                if (PlatformDetection.IsFullFramework)
```

Comment:
Do you plan to cleanup these `PlatformDetection.IsFullFramework` checks as well, or are they really needed?

Reply:
Please refer to 

```
// Note: ConvertFromString will raise exception if value cannot be converted.
values[i] = (int)intConverter.ConvertFromString(context, culture, tokens[i]);
```

Whatever exception ConvertFromString throw gets returned here, unless I catch and rethrow with Exception rather than ArgumentException to keep the behavior similar to full framework.

Reply:
If this is the same code as .NET Framework, why does it throw different exception? Is there a behavior change in IntConverter on .NET Core?

Reply:
For IntConverter we throw `Exception` in .NET Framework and `ArgumentException` for .NET Core. 

https://referencesource.microsoft.com/#System/compmod/system/componentmodel/basenumberconverter.cs,107

https://github.com/dotnet/corefx/blob/master/src/System.ComponentModel.TypeConverter/src/System/ComponentModel/BaseNumberConverter.cs#L84

I think since Exception is ArgumentException base class it is fine to leave it as is, right?

Reply:
Ok.

You can use `Throws(Type netCoreExceptionType, Type netFxExceptionType, Action action)` here to avoid the code duplication.